app/lib/report_helper.rb

```diff
@@ -169,6 +169,31 @@ def external_report_info(pipeline_run_id, background_id, params)
     data
   end
 
+  def label_top_scoring_taxa!(tax_map)
+    # Rule:
+    #   top 3 hits by NT.aggregatescore that satisfy:
+    #   NT.zscore > 1 AND
+    #   NR.zscore > 1 AND
+    #   NT.rpm > 1 AND
+    #   NR.rpm > 1 AND
+    #   tax_id > 0
+    # Current implementation assumes tax_map is already sorted by aggregate score.
+    # This assumption is valid because of the current value of DEFAULT_SORT_PARAM.
+    # All user-triggered sorting is implemented in the JSX frontend, so won't affect sorting of tax_map
+    # (despite some unused relics of code that accepted sort_by as a param in samples_controller).
+    top_taxa = []
+    i = 0
+    while top_taxa.length < 3
```

Comment:
Always define numbers as constants instead of using magic numbers.

Reply:
ok i made it an argument to the function instead since we might want to play with it in the future